### PR TITLE
drivers: nxp_enet: Improve readability within the driver

### PR DESCRIPTION
**Pull request by decsny**
Part of the issue with eth_mcux was the bloated source code making it slightly more difficult to maintain. A big part of the mess is already addressed by the initial merge point of the driver compared to eth_mcux, but some functions I did not clean up in the rx/tx code paths initially. So here I cleaned the rest of it up to make sure the community can test that for sure cleanup does not break anything for anyone (it should be functionally the same) before deprecating the old driver. 

This PR should make the source more readable and easy to understand, to avoid confusing code paths and conditional compilation messes that could lead to bugs being introduced.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM

File: drivers/ethernet/eth_nxp_enet.c [hunk 1]
```diff
@@ -157,105 +145,85 @@ static void net_if_mcast_cb(struct net_if *iface,
...
+	if (pkt_is_ptp) {
+		net_pkt_set_priority(pkt, NET_PRIORITY_CA);
+	}
 
-	return true;
+	return pkt_is_ptp;
 }
 
 
-#if defined(CONFIG_NET_L2_PTP)
```
Comment (mmahadevan108) [inline drivers/ethernet/eth_nxp_enet.c:204]: Why is this deleted?

File: drivers/ethernet/eth_nxp_enet.c [hunk 2]
```diff
@@ -658,9 +571,7 @@ static void eth_callback(ENET_Type *base, enet_handle_t *handle,
 		k_sem_give(&data->rx_thread_sem);
 		break;
 	case kENET_TxEvent:
-#if defined(CONFIG_PTP_CLOCK_NXP_ENET) && defined(CONFIG_NET_L2_PTP)
```
Comment (mmahadevan108) [inline drivers/ethernet/eth_nxp_enet.c:661]: Why is this deleted?

Comment (decsny) [reply drivers/ethernet/eth_nxp_enet.c:661]: it isnt needed

File: drivers/ethernet/eth_nxp_enet.c (hunk 1 above)
Comment (decsny) [reply drivers/ethernet/eth_nxp_enet.c:204]: CONFIG_PTP_CLOCK_NXP_ENET depends on CONFIG_NET_L2_PTP anyways

**Review by mmahadevan108 — APPROVED** [review]
